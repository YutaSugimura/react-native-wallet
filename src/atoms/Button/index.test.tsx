import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from './index';

describe('renders correctly', () => {
  it('atoms/Button (default)', () => {
    const tree = renderer.create(<Button text="button click" onPress={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('atoms/Button (props)', () => {
    const tree = renderer
      .create(
        <Button
          text="button click"
          onPress={() => {}}
          width={360}
          height={60}
          backgroundColor="#1C1C1C"
          pressedBackgroundColor="#1C1C1C"
          borderColor="#1C1C1C"
          borderWidth={1}
          borderRadius={12}
          fontColor="#1C1C1C"
          fontSize={15}
          fontWeight="700"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
