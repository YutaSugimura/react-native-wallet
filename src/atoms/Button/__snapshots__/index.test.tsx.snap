// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly atoms/Button (default) 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#FFF",
        "borderColor": undefined,
        "borderRadius": 0,
        "borderWidth": 1,
        "height": 46,
        "width": 300,
      },
      false,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#1C1C1C",
          "fontSize": 14,
          "fontWeight": "400",
        },
        Object {
          "color": undefined,
          "fontSize": 13,
          "fontWeight": "bold",
          "lineHeight": undefined,
        },
      ]
    }
  >
    button click
  </Text>
</View>
`;

exports[`renders correctly atoms/Button (props) 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#1C1C1C",
        "borderColor": "#1C1C1C",
        "borderRadius": 12,
        "borderWidth": 1,
        "height": 60,
        "width": 360,
      },
      false,
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#1C1C1C",
          "fontSize": 14,
          "fontWeight": "400",
        },
        Object {
          "color": "#1C1C1C",
          "fontSize": 15,
          "fontWeight": "700",
          "lineHeight": undefined,
        },
      ]
    }
  >
    button click
  </Text>
</View>
`;
